sdk/python/kfp/containers/_build_image_api.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the speci
+
+__all__ = [
+    'build_image_from_env',
+]
+
+
+import logging
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+import requests
+
+from . import get_default_image_builder
+from ..compiler._container_builder import ContainerBuilder
+
+
+def get_python_image() -> str:
```

Comment:
I wonder whether this is the best way.
The user clearly specifies the set of versioned packages they want using requirements.txt. It would be pretty strange to build a huge gigabyte+ image with many packages when the user only requested a few packages.
It might be less confusing if the users specifying the requirements, will be getting the exact packages they've asked for. 
What do you think?

Reply:
I think we should minimize the chance that user needs to change the requirements.txt file with a good base image. It should cover most of user's need. If user cares about small images, they are probably more advanced users who care about system efficiency. I am fine to let those users to customize the base image.

Reply:
I can do this, but we might be making some sacrifices when going with this.
By switching from dynamic python version to pinned version, we possibly reduce the compatibility with the user's environments. For example, if the user works in python 3.7 and uses [dataclasses](https://realpython.com/python37-new-features/#data-classes), their code will fail under python 3.6 which is used by the DL VM.

Reply:
I think it's fine. We might later build images based on DL image to support more python versions.

Reply:
Done.

The size of the DL image is 4.35GB which really concerns me. The GPU image size is 6.45GB.